src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -105,10 +106,15 @@
     private final AtomicReference<Future<?>> spoolingFuture = new AtomicReference<>();
     private final int maxInFlightPublishes = DEFAULT_MAX_IN_FLIGHT_PUBLISHES;
 
+    @Getter
     private final MqttClientConnectionEvents callbacks = new MqttClientConnectionEvents() {
         @Override
         public void onConnectionInterrupted(int errorCode) {
             setMqttOnline(false);
+            if (spoolingFuture.get() != null && !spoolingFuture.get().isCancelled()) {
+                spoolingFuture.get().cancel(true);
```

Comment:
Careful about how we handle this case - interrupting the publish loop now means that we need to make sure we don't lose a message. A different approach is to put the publish loop to sleep while the device is offline, rather than starting and stopping the task.